lib/bankaccounts/account.rb

```diff
@@ -0,0 +1,53 @@
+module BankAccounts
+
+  class Account
+    attr_accessor :id, :min_balance, :initial_balance, :balance, :fee
+
+    def initialize(id, initial_balance = 0)
+      @id = id
+      @min_balance = 0
+      @initial_balance = initial_balance
+      # Raises ArgumentError if initial_balance < 0
+      check_initial_balance(initial_balance)
+      @balance = initial_balance
+      @fee = 0
+    end
+
+    def check_initial_balance(initial_balance)
+      if initial_balance == nil || initial_balance < @min_balance
+        raise ArgumentError.new("Initial balance must be #{min_balance} or greater.")
+      end
+    end
+
+    # Subtracts amount from balance and returns balance
+    def withdraw(amount)
+      # amount and fee are subtracted from balance.
+      new_balance = @balance - amount - @fee
+
+      # warn if there's a fee
+      warn("Warning: This transaction will incur a $#{fee} fee.") if @fee > 0
```

Comment:
I like that you're using `warn` for these messages. It means we can redirect those messages to a separate stream for processing.